### DISCUSSION TITLE
Will you do buttons, and interaction receiving events?

### DISCUSSION THREAD

**Discussion by aprobeni** [discord-jda/JDA#1633] [Questions and Help] [answered]
Hello! I made buttons under a message with posting a request to the discord api. But i have one problem, and that is port forwarding. For some reason i can't ping my own IP, and because of this, if i can make a webserver to react to the interactions, it's useless, because even if i do the port forwarding, discord can't send me anything. 
It's a message's component, so it's basically an array in the "Message" class.
This is the documentation: 
https://discord.com/developers/docs/interactions/message-components
Receiving:
https://discord.com/developers/docs/interactions/slash-commands#receiving-an-interaction
It would really help, if it would be implemented, because it's a "discord bot thing" basically, and the interaction receiving is an event, if i am correct.

**Comment by MinnDevelopment** [comment, marked as answer]
https://github.com/DV8FromTheWorld/JDA/pull/1501
